### PR TITLE
[Exception] sales#update (Ekylibre::Record::RecordNotUpdateable) Reco…

### PR DESCRIPTION
**Pull request by pquintrie**
…rd cannot be updated

### CODE REVIEW

File: app/controllers/backend/base_controller.rb [hunk 1]
```diff
@@ -129,7 +129,12 @@ def find_and_check(*args)
     def save_and_redirect(record, options = {})
       record.attributes = options[:attributes] if options[:attributes]
       ActiveRecord::Base.transaction do
-        if options[:saved] || record.send(:save)
+        can_be_saved = if record.new_record?
```
Comment (piotaixr) [inline app/controllers/backend/base_controller.rb]: Can maybe be simplified by 
`can_be_saved = (record.new_record? && record.createable?) || record.updateable?`

I dunno if it's really more readable, though. @Aquaj @Zank94 a thought on this?

File: app/controllers/backend/base_controller.rb [hunk 2]
```diff
@@ -129,7 +129,12 @@ def find_and_check(*args)
...
+                         record.createable?
+                       else
+                         record.updateable?
+                       end
+        if can_be_saved && (options[:saved] || record.send(:save))
```
Comment (piotaixr) [inline app/controllers/backend/base_controller.rb]: As the line is modified... `record.save` ?

File: app/controllers/backend/base_controller.rb (hunk 1 above)
Comment (Aquaj) [reply app/controllers/backend/base_controller.rb]: I think a ternary operator would work here, the `if/else` does feel a bit clunky, but the `(... && ...) || ...` feels hard to parse if you're not used to creative applications of lazy evaluation.
